lib/commands/publish.js

```diff
@@ -102,19 +102,20 @@ class Publish extends BaseCommand {
       logTar(pkgContents, { unicode })
     }
 
+    const resolved = npa.resolve(manifest.name, manifest.version)
+    const registry = npmFetch.pickRegistry(resolved, opts)
+    const creds = this.npm.config.getCredentialsByURI(registry)
+    const outputRegistry = replaceInfo(registry)
+    if (!creds.token && !creds.username) {
+      throw Object.assign(
+        new Error(`This command requires you to be logged in to ${outputRegistry}`), {
+          code: 'ENEEDAUTH',
+        }
+      )
+    }
```

Comment:
then perhaps just this part could be in the non-dry-run branch?

Reply:
@wraithgar do you have an opinion on if only the error should be thrown during non-dry-run? or if this line should be guarded too

```
const creds = this.npm.config.getCredentialsByURI(registry)
```